File: Chapitre2/exercice2.3.c
//créer 5 processus fils avec for et break sans synchronisation
#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>
#include <errno.h>


int main()
{
    pid_t fils[5];
    for(int i=1; i<=5 ; i++)
    {
        fils[i]=fork();
        if( fils[i] == -1)
        {
            perror("Erreur lors de la création d'un des processus");
            exit(1);
        }
        if(fils[i] == 0) //je suis dans le fils
        {
            printf("Je suis dans le fils[%d] : moi=%d,père=%d\n",i,getpid(),getppid());
            break;
        }
    }

    return 0;
}

File: Chapitre2/exercice2.3.py
import os
import sys

def main():
    fils = [None] * 6

    for i in range(1, 6):
        fils[i] = os.fork()

        if fils[i] == -1:
            print(f"Error creating process: {os.strerror(os.errno)}", file=sys.stderr)
            sys.exit(1)

        if fils[i] == 0:
            print(f"I am in fils[{i}] : me={os.getpid()}, father={os.getppid()}")
            break

if __name__ == "__main__":
    main()
